src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1481,12 +1481,7 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node, format
         if (!_optimization_attributes.use_onednn_impls)
             return impl_types::ocl;
 
-        for (auto& dep : node.get_dependencies()) {
-            if (dep->is_in_data_flow() && dep->get_preferred_impl_type() == impl_types::onednn) {
-                preferred_impl = impl_types::onednn;
-                break;
-            }
-        }
+        preferred_impl = impl_types::onednn;
```

Comment:
This is quite a big change. Let me check the performance before/after this fix for entire test set.